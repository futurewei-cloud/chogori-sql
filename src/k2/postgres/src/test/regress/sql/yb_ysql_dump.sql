CREATE TABLE tbl1 (a SERIAL, b INT);
INSERT INTO tbl1 (b) VALUES (100);

CREATE TABLE tbl2 (a SERIAL);

CREATE TABLE tbl3(a INT, b INT, PRIMARY KEY(a ASC));

CREATE TABLE tbl4 (a INT, b INT, PRIMARY KEY(a HASH, b));

CREATE TABLE tbl5 (a INT PRIMARY KEY, b INT, c INT);
INSERT INTO tbl5 (a, b, c) VALUES (4, 7, 16);

CREATE TABLE tbl6 (a INT, PRIMARY KEY(a HASH));

CREATE TABLE tbl7 (a INT, b INT, c INT, d INT, PRIMARY KEY(b HASH, c));

CREATE TABLE tbl8 (a INT, b INT, c INT, d INT, PRIMARY KEY(a HASH, d));
CREATE INDEX tbl8_idx ON tbl8 ((b,c) HASH);
CREATE INDEX tbl8_idx2 ON tbl8 (a HASH, b);

CREATE TABLE tbl9 (a INT, b INT, c INT, PRIMARY KEY((a,b) HASH));

CREATE TABLE tbl10 (a INT, b INT, c INT, d INT, PRIMARY KEY((a,c) HASH, b));

CREATE TABLE tbl11 (a INT, b INT, c INT, PRIMARY KEY(a DESC, b ASC));

CREATE TABLE tbl12 (a INT, b INT, c INT, d INT, PRIMARY KEY(a ASC, d DESC, c DESC));

CREATE TABLE tbl13 (a INT, b INT, c INT, d INT, PRIMARY KEY((b,c) HASH));

CREATE USER rls_user NOLOGIN;

CREATE TABLE rls_public(k INT PRIMARY KEY, v TEXT);
CREATE TABLE rls_private(k INT PRIMARY KEY, v TEXT);

GRANT ALL ON rls_public TO public;
GRANT SELECT ON rls_private TO rls_user;

ALTER TABLE rls_public ENABLE ROW LEVEL SECURITY;
ALTER TABLE rls_private ENABLE ROW LEVEL SECURITY;
ALTER TABLE rls_private FORCE ROW LEVEL SECURITY;

CREATE POLICY p1 ON rls_public FOR ALL TO PUBLIC USING (k % 2 = 0);
CREATE POLICY p2 ON rls_private FOR INSERT WITH CHECK (k % 2 = 1);
CREATE POLICY p3 ON rls_private FOR UPDATE USING (k % 2 = 1);
