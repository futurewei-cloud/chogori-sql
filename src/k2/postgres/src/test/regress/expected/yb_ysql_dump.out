--
-- YSQL database dump
--

-- Dumped from database version 11.2-YB-2.0.4.0-b0
-- Dumped by ysql_dump version 11.2-YB-2.0.4.0-b0

SET statement_timeout = 0;
SET lock_timeout = 0;
SET idle_in_transaction_session_timeout = 0;
SET client_encoding = 'UTF8';
SET standard_conforming_strings = on;
SELECT pg_catalog.set_config('search_path', '', false);
SET check_function_bodies = false;
SET client_min_messages = warning;
SET row_security = off;

SET default_tablespace = '';

SET default_with_oids = false;

--
-- Name: rls_private; Type: TABLE; Schema: public; Owner: yugabyte_test
--

CREATE TABLE public.rls_private (
    k integer NOT NULL,
    v text,
    PRIMARY KEY((k) HASH)
);

ALTER TABLE ONLY public.rls_private FORCE ROW LEVEL SECURITY;


ALTER TABLE public.rls_private OWNER TO yugabyte_test;

--
-- Name: rls_public; Type: TABLE; Schema: public; Owner: yugabyte_test
--

CREATE TABLE public.rls_public (
    k integer NOT NULL,
    v text,
    PRIMARY KEY((k) HASH)
);


ALTER TABLE public.rls_public OWNER TO yugabyte_test;

--
-- Name: tbl1; Type: TABLE; Schema: public; Owner: yugabyte_test
--

CREATE TABLE public.tbl1 (
    a integer NOT NULL,
    b integer
);


ALTER TABLE public.tbl1 OWNER TO yugabyte_test;

--
-- Name: tbl10; Type: TABLE; Schema: public; Owner: yugabyte_test
--

CREATE TABLE public.tbl10 (
    a integer NOT NULL,
    b integer NOT NULL,
    c integer NOT NULL,
    d integer,
    PRIMARY KEY((a, c) HASH, b ASC)
);


ALTER TABLE public.tbl10 OWNER TO yugabyte_test;

--
-- Name: tbl11; Type: TABLE; Schema: public; Owner: yugabyte_test
--

CREATE TABLE public.tbl11 (
    a integer NOT NULL,
    b integer NOT NULL,
    c integer,
    PRIMARY KEY(a DESC, b ASC)
);


ALTER TABLE public.tbl11 OWNER TO yugabyte_test;

--
-- Name: tbl12; Type: TABLE; Schema: public; Owner: yugabyte_test
--

CREATE TABLE public.tbl12 (
    a integer NOT NULL,
    b integer,
    c integer NOT NULL,
    d integer NOT NULL,
    PRIMARY KEY(a ASC, d DESC, c DESC)
);


ALTER TABLE public.tbl12 OWNER TO yugabyte_test;

--
-- Name: tbl13; Type: TABLE; Schema: public; Owner: yugabyte_test
--

CREATE TABLE public.tbl13 (
    a integer,
    b integer NOT NULL,
    c integer NOT NULL,
    d integer,
    PRIMARY KEY((b, c) HASH)
);


ALTER TABLE public.tbl13 OWNER TO yugabyte_test;

--
-- Name: tbl1_a_seq; Type: SEQUENCE; Schema: public; Owner: yugabyte_test
--

CREATE SEQUENCE public.tbl1_a_seq
    AS integer
    START WITH 1
    INCREMENT BY 1
    NO MINVALUE
    NO MAXVALUE
    CACHE 1;


ALTER TABLE public.tbl1_a_seq OWNER TO yugabyte_test;

--
-- Name: tbl1_a_seq; Type: SEQUENCE OWNED BY; Schema: public; Owner: yugabyte_test
--

ALTER SEQUENCE public.tbl1_a_seq OWNED BY public.tbl1.a;


--
-- Name: tbl2; Type: TABLE; Schema: public; Owner: yugabyte_test
--

CREATE TABLE public.tbl2 (
    a integer NOT NULL
);


ALTER TABLE public.tbl2 OWNER TO yugabyte_test;

--
-- Name: tbl2_a_seq; Type: SEQUENCE; Schema: public; Owner: yugabyte_test
--

CREATE SEQUENCE public.tbl2_a_seq
    AS integer
    START WITH 1
    INCREMENT BY 1
    NO MINVALUE
    NO MAXVALUE
    CACHE 1;


ALTER TABLE public.tbl2_a_seq OWNER TO yugabyte_test;

--
-- Name: tbl2_a_seq; Type: SEQUENCE OWNED BY; Schema: public; Owner: yugabyte_test
--

ALTER SEQUENCE public.tbl2_a_seq OWNED BY public.tbl2.a;


--
-- Name: tbl3; Type: TABLE; Schema: public; Owner: yugabyte_test
--

CREATE TABLE public.tbl3 (
    a integer NOT NULL,
    b integer,
    PRIMARY KEY(a ASC)
);


ALTER TABLE public.tbl3 OWNER TO yugabyte_test;

--
-- Name: tbl4; Type: TABLE; Schema: public; Owner: yugabyte_test
--

CREATE TABLE public.tbl4 (
    a integer NOT NULL,
    b integer NOT NULL,
    PRIMARY KEY((a) HASH, b ASC)
);


ALTER TABLE public.tbl4 OWNER TO yugabyte_test;

--
-- Name: tbl5; Type: TABLE; Schema: public; Owner: yugabyte_test
--

CREATE TABLE public.tbl5 (
    a integer NOT NULL,
    b integer,
    c integer,
    PRIMARY KEY((a) HASH)
);


ALTER TABLE public.tbl5 OWNER TO yugabyte_test;

--
-- Name: tbl6; Type: TABLE; Schema: public; Owner: yugabyte_test
--

CREATE TABLE public.tbl6 (
    a integer NOT NULL,
    PRIMARY KEY((a) HASH)
);


ALTER TABLE public.tbl6 OWNER TO yugabyte_test;

--
-- Name: tbl7; Type: TABLE; Schema: public; Owner: yugabyte_test
--

CREATE TABLE public.tbl7 (
    a integer,
    b integer NOT NULL,
    c integer NOT NULL,
    d integer,
    PRIMARY KEY((b) HASH, c ASC)
);


ALTER TABLE public.tbl7 OWNER TO yugabyte_test;

--
-- Name: tbl8; Type: TABLE; Schema: public; Owner: yugabyte_test
--

CREATE TABLE public.tbl8 (
    a integer NOT NULL,
    b integer,
    c integer,
    d integer NOT NULL,
    PRIMARY KEY((a) HASH, d ASC)
);


ALTER TABLE public.tbl8 OWNER TO yugabyte_test;

--
-- Name: tbl9; Type: TABLE; Schema: public; Owner: yugabyte_test
--

CREATE TABLE public.tbl9 (
    a integer NOT NULL,
    b integer NOT NULL,
    c integer,
    PRIMARY KEY((a, b) HASH)
);


ALTER TABLE public.tbl9 OWNER TO yugabyte_test;

--
-- Name: tbl1 a; Type: DEFAULT; Schema: public; Owner: yugabyte_test
--

ALTER TABLE ONLY public.tbl1 ALTER COLUMN a SET DEFAULT nextval('public.tbl1_a_seq'::regclass);


--
-- Name: tbl2 a; Type: DEFAULT; Schema: public; Owner: yugabyte_test
--

ALTER TABLE ONLY public.tbl2 ALTER COLUMN a SET DEFAULT nextval('public.tbl2_a_seq'::regclass);


--
-- Data for Name: rls_private; Type: TABLE DATA; Schema: public; Owner: yugabyte_test
--

COPY public.rls_private (k, v) FROM stdin;
\.


--
-- Data for Name: rls_public; Type: TABLE DATA; Schema: public; Owner: yugabyte_test
--

COPY public.rls_public (k, v) FROM stdin;
\.


--
-- Data for Name: tbl1; Type: TABLE DATA; Schema: public; Owner: yugabyte_test
--

COPY public.tbl1 (a, b) FROM stdin;
1	100
\.


--
-- Data for Name: tbl10; Type: TABLE DATA; Schema: public; Owner: yugabyte_test
--

COPY public.tbl10 (a, b, c, d) FROM stdin;
\.


--
-- Data for Name: tbl11; Type: TABLE DATA; Schema: public; Owner: yugabyte_test
--

COPY public.tbl11 (a, b, c) FROM stdin;
\.


--
-- Data for Name: tbl12; Type: TABLE DATA; Schema: public; Owner: yugabyte_test
--

COPY public.tbl12 (a, b, c, d) FROM stdin;
\.


--
-- Data for Name: tbl13; Type: TABLE DATA; Schema: public; Owner: yugabyte_test
--

COPY public.tbl13 (a, b, c, d) FROM stdin;
\.


--
-- Data for Name: tbl2; Type: TABLE DATA; Schema: public; Owner: yugabyte_test
--

COPY public.tbl2 (a) FROM stdin;
\.


--
-- Data for Name: tbl3; Type: TABLE DATA; Schema: public; Owner: yugabyte_test
--

COPY public.tbl3 (a, b) FROM stdin;
\.


--
-- Data for Name: tbl4; Type: TABLE DATA; Schema: public; Owner: yugabyte_test
--

COPY public.tbl4 (a, b) FROM stdin;
\.


--
-- Data for Name: tbl5; Type: TABLE DATA; Schema: public; Owner: yugabyte_test
--

COPY public.tbl5 (a, b, c) FROM stdin;
4	7	16
\.


--
-- Data for Name: tbl6; Type: TABLE DATA; Schema: public; Owner: yugabyte_test
--

COPY public.tbl6 (a) FROM stdin;
\.


--
-- Data for Name: tbl7; Type: TABLE DATA; Schema: public; Owner: yugabyte_test
--

COPY public.tbl7 (a, b, c, d) FROM stdin;
\.


--
-- Data for Name: tbl8; Type: TABLE DATA; Schema: public; Owner: yugabyte_test
--

COPY public.tbl8 (a, b, c, d) FROM stdin;
\.


--
-- Data for Name: tbl9; Type: TABLE DATA; Schema: public; Owner: yugabyte_test
--

COPY public.tbl9 (a, b, c) FROM stdin;
\.


--
-- Name: tbl1_a_seq; Type: SEQUENCE SET; Schema: public; Owner: yugabyte_test
--

SELECT pg_catalog.setval('public.tbl1_a_seq', 1, true);


--
-- Name: tbl2_a_seq; Type: SEQUENCE SET; Schema: public; Owner: yugabyte_test
--

SELECT pg_catalog.setval('public.tbl2_a_seq', 1, false);


--
-- Name: tbl8_idx; Type: INDEX; Schema: public; Owner: yugabyte_test
--

CREATE INDEX tbl8_idx ON public.tbl8 USING lsm ((b, c) HASH);


--
-- Name: tbl8_idx2; Type: INDEX; Schema: public; Owner: yugabyte_test
--

CREATE INDEX tbl8_idx2 ON public.tbl8 USING lsm (a HASH, b);


--
-- Name: rls_public p1; Type: POLICY; Schema: public; Owner: yugabyte_test
--

CREATE POLICY p1 ON public.rls_public USING (((k % 2) = 0));


--
-- Name: rls_private p2; Type: POLICY; Schema: public; Owner: yugabyte_test
--

CREATE POLICY p2 ON public.rls_private FOR INSERT WITH CHECK (((k % 2) = 1));


--
-- Name: rls_private p3; Type: POLICY; Schema: public; Owner: yugabyte_test
--

CREATE POLICY p3 ON public.rls_private FOR UPDATE USING (((k % 2) = 1));


--
-- Name: rls_private; Type: ROW SECURITY; Schema: public; Owner: yugabyte_test
--

ALTER TABLE public.rls_private ENABLE ROW LEVEL SECURITY;

--
-- Name: rls_public; Type: ROW SECURITY; Schema: public; Owner: yugabyte_test
--

ALTER TABLE public.rls_public ENABLE ROW LEVEL SECURITY;

--
-- Name: TABLE rls_private; Type: ACL; Schema: public; Owner: yugabyte_test
--

GRANT SELECT ON TABLE public.rls_private TO rls_user;


--
-- Name: TABLE rls_public; Type: ACL; Schema: public; Owner: yugabyte_test
--

GRANT ALL ON TABLE public.rls_public TO PUBLIC;


--
-- YSQL database dump complete
--

